uwblueprint/starter-code-v2
# Fix various bugs

## Notion ticket link

Closes #110 



## Implementation description
* See commit messages



## Steps to test
1. 266fae7e53d4fa2217ec87b9b631cfe2c6427d73: check that creating or updating an entity  without supplying `enum_field` when using a Python REST backend does not raise an exception (it should return 400 bad request status successfully)
2. 42443b1f6c06270a11f93ba9094e0b2e414a970e: verify that Python backend tests run successfully
```bash
docker exec -it starter-code-v2_py-backend_1 /bin/bash -c "pip install -e . && pytest"
```
3. 5064ccaad15d365cd0c84ee9456991fd86bf1744: check that the Python server can connect to the PostgreSQL database successfully, make some API requests
4. 613c047cbc567b6bc181850276c6d818035a2b43: verify that TypeScript backend tests run successfully
```bash
docker exec -it starter-code-v2_ts-backend_1 /bin/bash -c "yarn test"
```
5. 0b34f2014489f06c3634188f313a6b0fcdf6329d: run the frontend with the REST client implementation in DisplayTableContainer.tsx, verify that the "Display Entities" page can successfully display data
6. 6b9e518ceba377eca5cef710e419017cc280b823: ensure that CSV download works with a Python backend when there is a mixture of entities with/without `file_name` field



## What should reviewers focus on?
* Review commit by commit, each commit addresses a different issue


## Checklist
